app/com/arpnetworking/metrics/portal/reports/impl/JobExecutor.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2018 Dropbox, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.arpnetworking.metrics.portal.reports.impl;
+
+import akka.actor.AbstractActor;
+import akka.actor.ActorRef;
+import akka.actor.Props;
+import com.arpnetworking.metrics.portal.reports.Job;
+
+/**
+ * An actor that executes a {@link Job}, i.e. renders it and sends out the resultant Report.
+ *
+ * @author Spencer Pearson
+ */
+public final class JobExecutor extends AbstractActor {
+
+    /**
+     * Props factory.
+     *
+     * @return A new Props to create this actor.
+     */
+    public static Props props() {
+        return Props.create(JobExecutor.class, JobExecutor::new);
+    }
+
+    @Override
+    public Receive createReceive() {
+        return receiveBuilder()
+                .match(Execute.class, e -> {
+                    e.getJob().getSpec().render()
```

Comment:
Still think this should push the specific action down to the job implementation instead of having it in the executor.